pkg/broker/ingress/server_manager.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2023 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ingress
+
+import (
+	"context"
+	"crypto/tls"
+
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/types"
+	"knative.dev/eventing/pkg/eventingtls"
+	"knative.dev/eventing/pkg/kncloudevents"
+	"knative.dev/pkg/configmap"
+
+	kubeclient "knative.dev/pkg/client/injection/kube/client"
+	secretinformer "knative.dev/pkg/injection/clients/namespacedkube/informers/core/v1/secret"
+)
+
+func NewServerManager(ctx context.Context, logger *zap.Logger, cmw configmap.Watcher, httpPort, httpsPort int, handler *Handler) (*eventingtls.ServerManager, error) {
+	tlsConfig, err := getServerTLSConfig(ctx)
+	if err != nil {
+		logger.Info("failed to get TLS server config", zap.Error(err))
+	}
+
+	httpReceiver := kncloudevents.NewHTTPMessageReceiver(httpPort)
+	httpsReceiver := kncloudevents.NewHTTPMessageReceiver(httpsPort, kncloudevents.WithTLSConfig(tlsConfig))
+
+	return eventingtls.NewServerManager(ctx, httpReceiver, httpsReceiver, handler, cmw)
+}
+
+func getServerTLSConfig(ctx context.Context) (*tls.Config, error) {
+	secret := types.NamespacedName{
+		Namespace: "knative-eventing",
+		Name:      "mt-broker-ingress-server-tls",
```

Comment:
Nit: can we introduce a constant somewhere for this secret name?

Reply:
@Cali0707 There are other secrets as well, such as the one for the filter.
Should we add constants for them in eventing-tls?

Maybe we should make a separate PR as I would like this PR to only contain ingress changes

Reply:
Let's record an issue for this improvement